ui/pages/permissions-connect/flask/util.js

```diff
@@ -13,60 +14,91 @@ import {
 } from '../../../helpers/utils/util';
 
 export function getSnapInstallWarnings(permissions, targetSubjectMetadata, t) {
-  const bip32EntropyPermissions =
-    permissions &&
-    Object.entries(permissions)
-      .filter(([key]) => key === 'snap_getBip32Entropy')
-      .map(([, value]) => value);
+  const bip32EntropyPermission = permissions.snap_getBip32Entropy;
```

Comment:
Could you do this in a loop instead, by looking at all permissions and getting their weight from `PERMISSION_DESCRIPTIONS`?

Reply:
I thought about that, but I'm not sure it makes sense, because we aren't rendering the descriptions in an abstract way. Grabbing the permissions from their weights doesn't mean much when we're writing custom logic for each weight-1 permission (which doesn't seem like it will change either because we don't know what future weight-1 permissions will be).

Reply:
How about using the actual return value from `PERMISSION_DESCRIPTIONS[permission](...)`? You could add a new field there.